src/lib/Scenes/Artwork/Components/ImageCarousel/ImageCarouselContext.tsx

```diff
@@ -0,0 +1,104 @@
+import { Schema } from "lib/utils/track"
+import { observable } from "mobx"
+import React, { useMemo, useRef } from "react"
+import { FlatList, View } from "react-native"
+import { useTracking } from "react-tracking"
+
+export interface ImageDescriptor {
+  url: string
+  width: number
+  height: number
+}
+
+export type ImageCarouselAction =
+  | {
+      type: "IMAGE_INDEX_CHANGED"
+      nextImageIndex: number
+    }
+  | {
+      type: "TAPPED_TO_GO_FULL_SCREEN"
+    }
+  | {
+      type: "FULL_SCREEN_DISMISSED"
+    }
+  | {
+      type: "FULL_SCREEN_FINISHED_ENTERING"
+    }
+  | {
+      type: "FULL_SCREEN_INITIAL_RENDER_COMPLETED"
+    }
+  | {
+      type: "ZOOM_SCALE_CHANGED"
+      nextZoomScale: number
+    }
+
+export type FullScreenState = "none" | "doing first render" | "animating entry transition" | "entered"
+
+export interface ImageCarouselState {
+  imageIndex: number
+  fullScreenState: FullScreenState
+  isZoomedCompletelyOut: boolean
+}
+
+export interface ImageCarouselContext {
+  state: Readonly<ImageCarouselState>
+  images: ImageDescriptor[]
+  embeddedImageRefs: View[]
+  embeddedFlatListRef: React.RefObject<FlatList<any>>
+  dispatch(action: ImageCarouselAction): void
+}
+
+export function useNewImageCarouselContext({ images }: { images: ImageDescriptor[] }): ImageCarouselContext {
+  const embeddedImageRefs = useMemo(() => [], [])
+  const embeddedFlatListRef = useRef<FlatList<any>>()
+  const state = observable({
+    imageIndex: 0,
+    fullScreenState: "none" as FullScreenState,
+    isZoomedCompletelyOut: true,
+  })
+  const tracking = useTracking()
+
+  return useMemo(
+    () => ({
+      state,
+      images,
+      embeddedImageRefs,
+      embeddedFlatListRef,
+      dispatch: (action: ImageCarouselAction) => {
```

Comment:
Why manually create `dispatch` vs using `useReducer`?

Reply:
`useReducer` is a way to update immutable state, but I opted for mutable state with `mobx`. it doesn't really make sense to mix the two.